apps/charterafrica/migrations/20230717085545-update_blocks_rename_collection_to_item.js

```diff
@@ -0,0 +1,71 @@
+module.exports = {
+  async up(db) {
+    const bulkOps = [];
+
+    await db
+      .collection("pages")
+      .find()
+      .forEach((doc) => {
+        if (doc.blocks && doc.blocks.length > 0) {
+          doc.blocks.forEach((block) => {
+            if (block.items && block.items.length > 0) {
+              block.items.forEach((item) => {
+                if (item.collection) {
+                  // eslint-disable-next-line no-param-reassign
+                  item.item = item.collection;
+                  // eslint-disable-next-line no-param-reassign
+                  delete item.collection;
+                  const { _id } = doc;
+                  bulkOps.push({
+                    updateOne: {
+                      filter: { _id },
+                      update: { $set: { blocks: doc.blocks } },
+                    },
+                  });
+                }
+              });
+            }
+          });
+        }
+      });
+
+    if (bulkOps.length > 0) {
+      db.collection("pages").bulkWrite(bulkOps);
+    }
+  },
+
+  async down(db) {
+    const bulkOps = [];
+
+    await db
+      .collection("pages")
+      .find()
+      .forEach((doc) => {
+        if (doc.blocks && doc.blocks.length > 0) {
+          doc.blocks.forEach((block) => {
```

Comment:
here you have to use for of rather than foreach as foreach doesn't support blocking

Reply:
@saminegash not sure why you would propose a for...of over forEach when we do not need to break/continue out of loop.